src/transport/SessionHolder.cpp

```diff
@@ -72,11 +74,24 @@ SessionHolder & SessionHolder::operator=(SessionHolder && that)
     return *this;
 }
 
+void SessionHolder::GrabPairing(const SessionHandle & session)
+{
+    Release();
+    if (session->IsSecureSession() && session->AsSecureSession()->IsPairing())
```

Comment:
The expectation of the function seems to be: "ensure the provided session is in the pairing state and grab it". Silently doing nothing is the session is not in pairing doesn't seem right.

Perhaps we could return a bool that indicates if the session was indeed grabbed and verify that?

That applies to both ::Grabs here.

Reply:
Done